AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java

```diff
@@ -291,13 +293,31 @@ private AndroidManifest parse(File androidManifestFile, boolean libraryProject)
 		componentQualifiedNames.addAll(receiverQualifiedNames);
 		componentQualifiedNames.addAll(providerQualifiedNames);
 
+		Map<String, String> metaDataQualifiedNames = new HashMap<String, String>();
+		NodeList metaDataNodes = documentElement.getElementsByTagName("meta-data");
+		for (int i = 0; i < metaDataNodes.getLength(); i++) {
+			Node node = metaDataNodes.item(i);
+			Node nameAttribute = node.getAttributes().getNamedItem("android:name");
+			Node valueAttribute = node.getAttributes().getNamedItem("android:value");
+
+			if (nameAttribute == null || valueAttribute == null) {
```

Comment:
Hi, I will make a review, and I will check directly the docs https://developer.android.com/guide/topics/manifest/meta-data-element.html, in General, I solved it to the needs of collecting the <meta-data> tags from within the manifest, and following the general structure:

```xml
<meta-data android:name="zoo" android:value="@string/kangaroo" />
```

I used also a Key-Value (a Map) to store and read all the elements. Checking in the docs, there's also a 3rd attribute that can be written in the Manifest, android:resource, never use it before to be honest, but in the general case should be included.

Maybe return a `Map<String, SomeObject>`, with SomeObject being a pair Value, Resource?.

On the other hands, I'm collecting the Metadata at the root level (Application), but it can be applied to:

```xml
    <activity>
    <activity-alias>
    <application>
    <provider>
    <receiver>
    <service>
```

In a general scenario, a reference to the parent tag should be also stored.

Any suggestion about this?
.